core/src/main/java/greencity/controller/EventController.java

```diff
@@ -168,13 +171,22 @@ public ResponseEntity<EventDto> getEvent(
     })
     @ApiPageableWithoutSort
     @GetMapping
-    public ResponseEntity<PageableAdvancedDto<EventDto>> getEvent(
+    public ResponseEntity<PageableAdvancedDto<EventDto>> getEvents(
         @Parameter(hidden = true) Pageable pageable,
         @RequestParam(required = false, name = "user-id") Long userId,
         FilterEventDto filterEventDto) {
+        if (filterEventDto != null && filterEventDto.getStatuses() != null) {
+            validateStatusesRequireUserId(filterEventDto.getStatuses(), userId);
+        }
         return ResponseEntity.ok().body(eventService.getEvents(pageable, filterEventDto, userId));
     }
 
+    private void validateStatusesRequireUserId(List<EventStatus> statuses, Long userId) {
```

Comment:
Maybe it is better to transfer this logic in service layer

Reply:
No. This simple validation, that do not related to business logic, should be on controller layer. For example see DTO validation using annotation from jakarta.validation.constraints package.

Reply:
If I could write this with annotations I would, but this validation requires two objects: statuses list and userId.